src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = '';
+      }
+
+      result += element;
```

Comment:
This one looks better. We do action when we need, no need to overwrite to empty string
```suggestion
      const element = this[i];

      if (element !== null || element !== undefined) {
        result += element;
      }
```

Reply:
At least 4 tests are not working if I rewrite the code this way. So I changed it in a bit different way. Here is one of the unsuccessful tests - https://prnt.sc/ZqrOJ81I-c1S

Reply:
hmm, that sounds strange, can you please share a photo of your code after these changes?

Reply:
I'm so sorry, it seems that I missed something on the first try because on the second try everything went well. So it's my mistake, sorry.